source/fundamentals/monitoring.txt

```diff
@@ -1,7 +1,435 @@
+     - Created when the server is closed.
+
+   * - ``TopologyDescriptionChangedEvent``
+     - Created when the topology changes, such as an election of a new
+       primary or disconnection of a ``mongos`` proxy.
+
+   * - ``TopologyOpeningEvent``
+     - Created when the topology is initialized.
+
+   * - ``TopologyClosedEvent``
+     - Created when the topology is closed.
+
+   * - ``ServerHeartbeatStartedEvent``
+     - Created when the heartbeat is started.
+
+   * - ``ServerHeartbeatSucceededEvent``
+     - Created when the heartbeat succeeds.
+
+   * - ``ServerHeartbeatFailedEvent``
+     - Created when the heartbeat fails.
+
+Example Event Documents
+-----------------------
+
+The following sections show sample output for each type of SDAM event.
+
+ServerDescriptionChangedEvent
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: bash
+   :copyable: false
+      
+   *event.ServerDescriptionChangedEvent
+   {
+       "Address": "...",
+       "TopologyID": "...",
+       "PreviousDescription": {
+           "Addr": "...",
+           "Arbiters": null,
+           "AverageRTT": 0,
+           "AverageRTTSet": false,
+           "Compression": null,
+           "CanonicalAddr": "...",
+           "ElectionID": "...",
+           "HeartbeatInterval": 0,
+           "HelloOK": false,
+           "Hosts": null,
+           "LastError": null,
+           "LastUpdateTime": "...",
+           "LastWriteTime": "...",
+           "MaxBatchCount": 0,
+           "MaxDocumentSize": 0,
+           "MaxMessageSize": 0,
+           "Members": null,
+           "Passives": null,
+           "Passive": false,
+           "Primary": "...",
+           "ReadOnly": false,
+           "ServiceID": null,
+           "SessionTimeoutMinutes": 0,
+           "SetName": "...",
+           "SetVersion": 0,
+           "Tags": null,
+           "TopologyVersion": null,
+           "Kind": 0,
+           "WireVersion": null
+       },
+       "NewDescription": {
+           "Addr": "...",
+           "Arbiters": null,
+           "AverageRTT": ...,
+           "AverageRTTSet": true,
+           "Compression": null,
+           "CanonicalAddr": "...",
+           "ElectionID": "...",
+           "HeartbeatInterval": ...,
+           "HelloOK": true,
+           "Hosts": [...],
+           "LastError": null,
+           "LastUpdateTime": "...",
+           "LastWriteTime": "...",
+           "MaxBatchCount": ...,
+           "MaxDocumentSize": ...,
+           "MaxMessageSize": ...,
+           "Members": [...],
+           "Passives": null,
+           "Passive": false,
+           "Primary": "...",
+           "ReadOnly": false,
+           "ServiceID": null,
+           "SessionTimeoutMinutes": 30,
+           "SetName": "...",
+           "SetVersion": 9,
+           "Tags": [...],
+           "TopologyVersion": {...},
+           "Kind": 10,
+           "WireVersion": {...}
+       }
+   }
+
+The ``Kind`` field in the server descriptions represents the type of a
```

Comment:
Question: I agree that Kind should be documented here, but are the rest of the document fields straightforward enough that we don't need to document them too?